core/workflow/graft.go

```diff
@@ -31,27 +31,22 @@ import (
 )
 
 // Graft takes a root node, a path to a role in root, byte array with an existing role and appends this
-// role to the role in root where the path specifies..
-func Graft(root *yaml.Node, path string, toAdd []byte) (out []byte, err error)  {
-    roleToAdd, err := LoadWorkflow(toAdd)
-    if err != nil {
-        return nil, err
-    }
-
+// role asa child of the role in root where the path specifies.
+func Graft(root *yaml.Node, path string, toAdd []byte, graftedName string) (out []byte, err error)  {
     var parent *yaml.Node
-
     for _, step := range strings.Split(path, PATH_SEPARATOR) {
-        // FIXME: no return value
         _ = iterateNode(root, &parent, step)
     }
-
     if parent == nil {
         return nil, fmt.Errorf("specified path not found")
     }
 
-    // Not appending to root, only to a copy of root
-    parent.Content = append(parent.Content, roleToAdd.Content[0])
+    err = appendRole(parent, toAdd)
+    if err != nil {
+        return nil, err
+    }
 
+    root.Content[0].Content[1].Value = graftedName
```

Comment:
Hard-coding the path of `name` for the root role because ideally it should always be present at the same location. We could also perform a search through the `yaml.Node` using `iterateNode`.